### PR TITLE
updates to data mgt, plot tooltips, and focalplane scatter plot

### LINKED ISSUE #2: Add focalplane scatter plot
Add block of code for plotting values by positioner in the focalplane.

### LINKED ISSUE #3: Add data to positioner files and guider files
Additional data is needed for certain plots. I will add this data , but implementation of how the data is compiled in the future may change.

### LINKED ISSUE #4: Add to bokeh tools
Decide on set of bokeh tools for plots and add these

### PR DESCRIPTION
**Pull request by parfa30**
This pull request has changes for all code and adds the following:
* data_mgt/data_general.py: main functions for getting data from various places
* data_mgt/data_handler.py: includes functions to initialize and update all data files for plots
* static/plots.py: Using this instead of pages.py. Includes functions used by all pages. Used to standardize plot types.
* data/focalplane: data for the focalplane and guider plots
* data/detector: data for the detector plots
* data/fiberpos.csv: map of positioner location, spectrograph, and fiber number

### CODE REVIEW

File: py/desiperf/instperfapp/data_mgt/data_general.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+from astropy.table import Table
+from astropy.time import Time
+import psycopg2
+from datetime import datetime
+from bokeh.models import ColumnDataSource
+
+
+class DataSource():
+    # Currently this only works on the DESI Server
```
Comment (paulmartini) [inline py/desiperf/instperfapp/data_mgt/data_general.py:15]: Future upgrade: Add a try/except loop, probably at psycopg2.connect(), to handle cases where someone attempts to run a DB query when not on the DESI server

File: py/desiperf/instperfapp/pages/focalplane.py [hunk 2]
```diff
@@ -48,22 +49,25 @@ def get_data(self, attr1, attr2, update=False):
...
                               [self.ts2]])
-        tab = Panel(child=this_layout, title=self.page.title)
+        tab = Panel(child=this_layout, title=self.plots.title)
         return tab
 
     def time_series_plot(self):
-        self.corr = figure(plot_width=350, plot_height=250, tools=self.page.tools, x_axis_label=self.x_select.value, y_axis_label=self.y_select.value)
-        self.ts1 = figure(plot_width=900, plot_height=200, tools=self.page.tools, active_drag="xbox_select", x_axis_label='MJD_OBS', y_axis_label=self.x_select.value)
-        self.ts2 = figure(plot_width=900, plot_height=200, tools=self.page.tools, active_drag="xbox_select", x_axis_label='MJD_OBS', y_axis_label=self.y_select.value)
+        self.corr = self.plots.figure(width=350, height=250, x_axis_label='attr1', y_axis_label='attr2')
```
Comment (paulmartini) [inline py/desiperf/instperfapp/pages/focalplane.py:66]: I'm not sure why you switched to the labels 'attr1' and 'attr2' - I don't find those that informative!

Comment (parfa30) [reply py/desiperf/instperfapp/pages/focalplane.py:66]: I agree. I think we will want to change this in the future. My problem at this point is that I can't get the x_axis_label to change when they are re-selected.

File: py/desiperf/instperfapp/pages/positioner.py [hunk 3]
```diff
@@ -2,52 +2,79 @@
...
-        self.btn = self.page.button('Get Pos')
+        self.btn = Button(label = 'Get Pos', button_type='primary',width=300)
 
 
     def page_layout(self):
-        this_layout = layout([[self.page.header],
-                        [self.pos_select, self.x_select, self.btn],
-                        [self.ts1]])
-        tab = Panel(child=this_layout, title=self.page.title)
+        this_layout = layout([[self.plots.header],
```
Comment (paulmartini) [inline py/desiperf/instperfapp/pages/positioner.py:43]: Upgrade suggestion: I think the scatter plot should go at the top (just below the pull-down menus), perhaps next to the correlation plot.

File: py/desiperf/instperfapp/data_mgt/data_general.py (hunk 1 above)
Comment (parfa30) [reply py/desiperf/instperfapp/data_mgt/data_general.py:15]: I will put this in now and re-commit. I can also add a message to alert the user why it won't work for them.

**Review by paulmartini — APPROVED** [review]

File: py/desiperf/instperfapp/data_mgt/data_general.py [hunk 4]
```diff
@@ -1,5 +1,6 @@
 
 import pandas as pd
+import fnmatch
```
Comment (paulmartini) [inline py/desiperf/instperfapp/data_mgt/data_general.py:3]: I think you should avoid adding this dependency just to keep the total list of required modules small.

Comment (parfa30) [reply py/desiperf/instperfapp/data_mgt/data_general.py:3]: Ok, that shouldn't be difficult

File: py/desiperf/instperfapp/pages/focalplane.py (hunk 2 above)
Comment (paulmartini) [reply py/desiperf/instperfapp/pages/focalplane.py:66]: I ran into that too. Meant to flag it as a problem.

**Review by paulmartini — APPROVED** [review]

**Pull request merged**